exovetter/tce.py

```diff
@@ -1,85 +1,100 @@
-            raise ValueError(f'The period ({self.period}) is shorter than '
-                             f'the duration ({self.duration}).')
-
-    def get_boxmodel(self, times):
-        """Return box model, which is also stored in ``self.model``."""
-
-        offset = 0.5  # This offset is not returned to user.
-        model = np.ones(len(times))
-
-        self.model_times = times
-        self.model_phases = np.fmod(
-            times - self.tzero + (offset * self.period), self.period)
-
-        phases = self.model_phases
-
-        want = ((phases > offset * self.period - 0.5 * self.duration / 24) &
-                (phases <= offset * self.period + 0.5 * self.duration / 24))
-
-        model[want] = -1 * self.depth
-
-        # plt.figure()
-        # plt.plot(phases, model,'.')
-        # plt.title('tce box model')
-
-        self.model = model
-
-        return model
+# -*- coding: utf-8 -*-
+
+"""
+A TCE class stores the measured properties of a proposed transit.
+Those properties include orbital period, transit depth etc.
+
+This class tries to reduce the risk of modeling a transit
+with data in the wrong unit, for example, entering the transit duration in
+hours, then treating the value as if it is the duration in days.
+
+A Tce class is a dictionary with a list of reserved keywords. The
+values of these keys must be astropy.units.Quantities objects. Other
+keys in the dictionary have no restrictions on their values. By ensuring
+that certain measured quantities are have included units, we can ensure
+that, for example, a TCE created with depth measured in parts per million,
+is used correctly in code that expects depths to be measured in fractional
+amplitude.
+
+Transit times represent a wrinkle in this model. Most transit data has
+times corrected to the barycentre of the solar system, and expressed in
+units of days since some zero point. However, there is no agreement
+on the zero point. Some data is given in Julian Date, other missions choose
+mission specific zero points. For example, t=0 for Kepler data corresponds
+to a barycentric julian date of 2,454,833.0. Some common offsets are stored
+in const.py
+
+The Tce class addresses these zero points by making `epoch_offset` a
+reserved keyword. When creating a Tce, you must specify the period and
+epoch of the transit (typically with units of days), but also the time
+of the zero point of the time system.
+
+Example
+----------
+::
+
+    period_days = 5.3
+    epoch_days = 133.4
+    tce = Tce(period=periods_days * u.day,
+              epoch=epoch_days * u.day,
+              epoch_offset=const.bkjd)
+
+
+You can retrieve the epoch of the transit with the `get_epoch()` method.::
+
+    # Even though the Tce is created with transit time in BKJD, getting the
+    # Julian date of the transit is easy:
+    epoch_bjd = tce.get_epoch(const.bjd)
+
+See model.py for code to create model transits from a Tce object.
+"""
+import astropy.units as u
+
+
+class Tce(dict):
```

Comment:
A bit late to the discussion but I don't see the reason to rename `TCE` back to `Tce`. TCE is an acronym.